winrt/lib/xaml/BaseControl.h

```diff
@@ -712,6 +714,8 @@ namespace ABI { namespace Microsoft { namespace Graphics { namespace Canvas { na
         // be set up immediately when the object is constructed.
         void RegisterEventHandlersOnSelf()
         {
+            GetActivationFactory(Wrappers::HStringReference(RuntimeClass_Windows_UI_Xaml_Media_VisualTreeHelper).Get(), m_VisualTreeHelper.GetAddressOf());
```

Comment:
I don't see why you'd need this stored. Let's just request when we need it.

Reply:
This seems like an expensive function.

Reply:
How does it seem to be expensive? What evidence you have that is expensive? It's just a com getter, should be fast. It's not appropriate to store that pointer on the Control class, since it is not part of the state of the control. Also, for most scenarios, Loaded/Unloaded will only be called once, so no need to increase the control size for caching this  value.